image_captioning/models/encoder/encoder.py

```diff
@@ -0,0 +1,47 @@
+"""Encoder part of the image captioning module
+"""
+from typing import Iterable, Union
+
+import tensorflow as tf
+
+from image_captioning.models.encoder.model_selector import select_model
+
+
+class Encoder(tf.keras.Model):
+    """Encoder wrapper for encoders of different backbones
+    """
+
+    def __init__(self,
+                 model_name: str = 'inception_v3',
+                 embedding_dim: int = 64):
+        super().__init__()
+        self.model_name = model_name
+        self.embedding_dim = embedding_dim
+
+        self.model = select_model(model_name, embedding_dim)
+
+    def call(self,
+             images: Union[Iterable[tf.Tensor], tf.Tensor],
+             training=None,
+             mask=None) -> tf.Tensor:
+        return self.model(images)
+
+    def get_config(self):
+        return {
+            'model_name': self.model_name,
+            'embedding_dim': self.embedding_dim
+        }
+
+
+if __name__ == '__main__':
+    # Test Encoder
```

Comment:
Please remove these from non-script modules or put these in a separate test file: `encoder_test.py`.

Reply:
In the same folder or in a separate directory?

Reply:
I personally prefer in the same directory so that it is easily seen. If we use another directory e.g. `tests/`, it is not clear whether we have a test file for a certain module. We just need to be careful not to name the module `.*_test.py`.

For framework I think the standard's [unittest](https://docs.python.org/3/library/unittest.html#module-unittest) is fine, but I'm open to any better alternatives.

Reply:
I'm unfamiliar with unittest nor the framework you are talking about, but I'll make some changes and see what you think.

Reply:
You can see some of the examples in #14, but it is ok to not implement unit tests at all. This is not for corp product after all :)